### PR TITLE
Sequential Module Train to Eval mode update

### LINKED ISSUE #1426: Sequential.eval() does not put model into eval mode
Calling `eval()` should make `training` false. However, this does not work for `Sequential` modules.

Example F# program:

```fsharp
open type TorchSharp.torch.nn

let linear = Linear(10, 10)
linear.eval()
assert(not linear.training)       // succeeds

let sequential = Sequential(Linear(10, 10))
sequential.eval()
assert(not sequential.training)   // fails
```

### PR DESCRIPTION
**Pull request by alinpahontu2912**
Fixes #1426

### CODE REVIEW

File: RELEASENOTES.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ Releases, starting with 9/2/2021, are listed with the most recent release at the
 
 Move to libtorch 2.5.1. As with the 2.4.0 release, MacOS / Intel is no longer supported by libtorch, so TorchSharp doesn, either.
 
+__Bug Fixes__:
+
+#1426 Sequential.eval() does not put model into eval mode<br/>
```
[the remark this answers is not in the document]
Comment (alinpahontu2912) [reply RELEASENOTES.md]: Added new section, but did not change version strings for now.
